pkg/speccheck/speccheck.go

```diff
@@ -100,7 +101,10 @@ func (sc *SpecCheck) compareActualVsSpec(spec, actual *unstructured.Unstructured
 	updated.SetLabels(spec.GetLabels())
 	updated.SetAnnotations(processAnnotations(spec.GetAnnotations(), updated.GetAnnotations()))
 	updated.SetOwnerReferences(spec.GetOwnerReferences()) // TODO Is this ok? Check that there is only one controller and it is THIS bundle
-	updated.SetFinalizers(spec.GetFinalizers())           // TODO Is this ok?
+
+	finalizers := sets.NewString(updated.GetFinalizers()...)
+	finalizers.Insert(spec.GetFinalizers()...)
+	updated.SetFinalizers(finalizers.List()) // Sorted list of all finalizers
```

Comment:
Service Catalog adds a finalizer but we are discarding it here. Spec-vs-actual check after update/create was catching this issue. This shows that nobody ran Service Catalog integration tests for a while - at least since #192.